src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -253,7 +245,7 @@ case class AppDefinition(
       if (proto.hasUnreachableStrategy)
         UnreachableStrategy.fromProto(proto.getUnreachableStrategy)
       else
-        UnreachableStrategy.default(residencyOption.isDefined)
+        UnreachableStrategy.default(isResident)
```

Comment:
Where is this coming from since you've removed the original. Is it because the original was overridden?